File: frontend/src/app/main/models/TestConfiguration.ts
import { HeaderVal } from '../models/Header';
import { FormVal } from '../models/FormVal';
import { isPending } from 'q';
import { FileDetails } from '../models/FileDetails';

export class TestConfiguration {

    id: string;
    userId: string;
    testName: string;
    testDescription: string;
    url: string;
    baseUrl: string;
    basePath: string;
    endpointAction: string;
    payloadBody: string;
    payloadHeaders: HeaderVal[];
    formContent: FormVal[];
    response: string;
    status: string;
    file: FileDetails;
    tenant: string;

    constructor(obj?: any) {
        this.id = obj && obj.id || null;
        this.userId = obj && obj.userId || null;
        this.testName = obj && obj.testName || null;
        this.testDescription = obj && obj.testDescription || null;
        this.url = obj && obj.url || null;
        this.baseUrl = obj && obj.baseUrl || null;
        this.basePath = obj && obj.basePath || null;
        this.endpointAction = obj && obj.endpointAction || null;
        this.payloadBody = obj && obj.payloadBody || null;
        this.payloadHeaders = obj && obj.payloadHeaders || null;
        this.formContent = obj && obj.formContent || null;
        this.response = obj && obj.response || null;
        this.status = obj && obj.status || null;
        this.file = obj && obj.file || null;
        this.tenant = obj && obj.tenant || this.getTenant();
    }

    getTenant() {
        if (this.baseUrl !== null) {
            const splitString = this.baseUrl.split('/');
            return splitString[2];
        } else {
            return null;
        }

    }
}

File: ConnectAPIBuddy/Models/TestConfiguration.cs
using Microsoft.AspNetCore.Http;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.XPath;


namespace ConnectAPIBuddy.Models
{
    public class TestConfiguration
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string UserId { get; set; }
        public string TestName { get; set; }
        public string TestDescription { get; set; }
        public string Url { get; set; }
        public string BaseUrl { get; set; }
        public string BasePath { get; set; }
        public string EndpointAction { get; set; }
        public string PayloadBody { get; set; }
        public HeaderVal[] PayloadHeaders { get; set;}
        public FormVal[] FormContent { get; set; }
        public string Response { get; set; }
        public FileDetails File { get; set; }  
        public string Tenant { get; set; }
        public string Status { get; set; }

        
    }

    public class HeaderVal
    {
        public string Header { get; set; }
        public string Value { get; set; }
    }

    public class FormVal
    {
        public string Key { get; set; }
        public string Value { get; set; }
    }

    public class FileDetails
    {
        public string Name { get; set; }
        public string Type { get; set; }
        public long Size { get; set; }
        public string FileAsBase64 { get; set; }
        public string Key { get; set; }
    }

}

